WooCommerce/WooCommerceTests/ViewRelated/Products/Add Product/AddProductWithAI/ProductCreationAIPromptProgressBarViewModelTests.swift

```diff
@@ -0,0 +1,141 @@
+import XCTest
+@testable import WooCommerce
+import WooFoundation
+import SwiftUI
+
+class ProductCreationAIPromptProgressBarViewModelTests: XCTestCase {
+
+    // MARK: - Test updateText(to:)
+    func test_update_text_to_empty_string() {
+        // Given
+        let viewModel = ProductCreationAIPromptProgressBarViewModel()
+
+        // When
+        viewModel.updateText(to: "")
+
+        // Then
+        XCTAssertEqual(viewModel.status, .start)
+    }
+
+    func test_update_text_to_five_words() {
+        // Given
+        let viewModel = ProductCreationAIPromptProgressBarViewModel()
+
+        // When
+        viewModel.updateText(to: "This is a test input.")
+
+        // Then
+        XCTAssertEqual(viewModel.status, .inProgress)
+    }
+
+    func test_update_text_to_fifteen_words() {
+        // Given
+        let viewModel = ProductCreationAIPromptProgressBarViewModel()
+
+        // When
+        viewModel.updateText(to: "This is a test input with fifteen words to check halfway status.")
+
+        // Then
+        XCTAssertEqual(viewModel.status, .halfway)
+    }
+
+    func test_update_text_to_twenty_five_words() {
+        // Given
+        let viewModel = ProductCreationAIPromptProgressBarViewModel()
+
+        // When
+        viewModel.updateText(to: "This is a test input with twenty five words to check the almost done status of the progress bar view model in Swift.")
+
+        // Then
+        XCTAssertEqual(viewModel.status, .almostDone)
+    }
+
+    func test_update_text_to_thirty_five_words() {
+        // Given
+        let viewModel = ProductCreationAIPromptProgressBarViewModel()
+
+        // When
+        viewModel.updateText(to: "This is a test input with thirty five words to check the completed status of the progress bar view model in Swift" +
+                             " Adding more words to ensure it exceeds thirty words.")
+
+        // Then
+        XCTAssertEqual(viewModel.status, .completed)
+    }
+
+    // MARK: - Test ProgressStatus Enum
+    func test_progress_status_start() {
+        // Given
+        let status = ProductCreationAIPromptProgressBarViewModel.ProgressStatus.start
+
+        // Then
+        XCTAssertEqual(status.progress, 0.03)
+        XCTAssertEqual(status.color, Color(uiColor: .gray(.shade50)))
+        XCTAssertEqual(status.mainDescription, "")
+        XCTAssertNotNil(status.secondaryDescription)
+        XCTAssertEqual(status.secondaryDescription, NSLocalizedString(
+            "productCreationAIPromptProgressBar.secondary.start",
```

Comment:
Nit: If we introduce a `Localization` nested enum, we can use it directly from the unit tests instead of duplicating the strings here. Applies to all tests here.

Reply:
Updated 👍 3fd7edb46fedf1e90ab9db8cc1b375cbeffae2fc